### ISSUE TITLE
Sequence Feature Extraction Pipeline - Sliding Window Extraction and Augmentation

### ISSUE THREAD

**Issue by dhar174** [dhar174/steel_defect_demo#67]
# Sequence Feature Extraction Pipeline

## Overview
Implement a robust pipeline to convert raw time-series sensor data into fixed-length sequences suitable for LSTM training. This includes sliding window sequence extraction, normalization, padding/truncation, data augmentation, and temporal-aware dataset splitting.

## File Location
- **Path**: `src/features/feature_extractor.py`
- **Dependencies**: `pandas`, `numpy`, `scikit-learn`

## Functional Requirements
1. **Sliding Window Extraction**
   - Extract overlapping sequences from continuous sensor data using configurable window size and stride.
2. **Normalization and Standardization**
   - Apply Z-score normalization per sensor channel.
   - Support optional Min-Max scaling if configured.
3. **Sequence Padding & Truncation**
   - Pad shorter sequences to fixed length with zeros.
   - Truncate longer sequences to the same fixed length.
4. **Data Augmentation Techniques**
   - Noise injection: add Gaussian noise to sequences.
   - Time warping: stretch/compress time axis randomly.
5. **Temporal Train/Validation/Test Splitting**
   - Split sequences into train, validation, and test sets preserving temporal order.
   - Support configurable split ratios and rolling-origin evaluation.

## Implementation Tasks
```python
class SequenceFeatureExtractor:
    def __init__(self, sequence_length: int, stride: int, augmentation_config: dict = None):
        """Initialize extractor with window and augmentation parameters"""
        pass

    def extract_sequences(self, data: pd.DataFrame) -> np.ndarray:
        """Sliding window sequence extraction"""
        pass

    def normalize_sequences(self, sequences: np.ndarray) -> np.ndarray:
        """Z-score normalization per feature channel"""
        pass

    def pad_sequences(self, sequences: np.ndarray) -> np.ndarray:
        """Pad or truncate sequences to fixed length"""
        pass

    def augment_sequences(self, sequences: np.ndarray) -> np.ndarray:
        """Apply noise injection and time warping"""
        pass

    def split_sequences(
                        sequences: np.ndarray,
                        labels: np.ndarray,
                        split_ratios: Tuple[float, float, float]) -> Tuple:
        """Temporal-aware train/val/test split"""
        pass
```

## Testing Requirements
- Unit tests for each method in `tests/test_feature_extractor.py`:
  - `test_extract_sequences()`
  - `test_normalize_sequences()`
  - `test_pad_sequences()`
  - `test_augment_sequences()`
  - `test_split_sequences()`
- Integration test with synthetic data to ensure end-to-end sequence preparation.

## Deliverables
- Completed `feature_extractor.py` with all required methods implemented
- Configuration support for window size, stride, and augmentation parameters
- Comprehensive unit and integration tests
- Documentation and usage examples in docstrings and README

## Dependencies & Environment
```bash
pip install pandas numpy scikit-learn
```

## Related Tasks
- Depends on: Phase 2: Feature Engineering (`CastingFeatureEngineer`)
- Enables: LSTM DataLoader & Training Pipeline (Tasks 3.3 and 3.4)

## Success Criteria
- Sequences correctly extracted and normalized
- Padding/truncation yields fixed-length arrays
- Augmentation methods produce valid variations
- Temporal splits respect data order without leakage
- All tests pass with ≥ 90% coverage
